manager/integration/tests/test_kubernetes_status.py

```diff
@@ -0,0 +1,225 @@
+    assert not k_status['LastPVCRefAt']
+    assert k_status['LastPodRefAt']
+
+    delete_and_wait_pvc_only(core_api, pod_info)
+    volume = client.by_id_volume(volume_name)
+    k_status = volume["kubernetesStatus"]
+    assert k_status['PVName'] == p['pv_name']
+    for i in range(RETRY_COMMAND_COUNT):
+        if k_status['PVStatus'] == 'Released':
+            break
+        time.sleep(1)
+        volume = client.by_id_volume(volume_name)
+        k_status = volume["kubernetesStatus"]
+    assert k_status['PVStatus'] == 'Released'
+    assert k_status['Namespace'] == 'default'
+    assert k_status['PVCName'] == p['pvc_name']
+    assert k_status['PodName'] == p['pod_name']
+    assert k_status['WorkloadName'] == statefulset_name
+    assert k_status['WorkloadType'] == 'StatefulSet'
+    assert k_status['LastPVCRefAt']
+    assert k_status['LastPodRefAt']
+
+    delete_and_wait_pv(core_api, pod_info)
+    volume = client.by_id_volume(volume_name)
+    k_status = volume["kubernetesStatus"]
+    assert k_status['PVName'] == ''
+    assert k_status['PVStatus'] == ''
+    assert k_status['Namespace'] == 'default'
+    assert k_status['PVCName'] == p['pvc_name']
+    assert k_status['PodName'] == p['pod_name']
+    assert k_status['WorkloadName'] == statefulset_name
+    assert k_status['WorkloadType'] == 'StatefulSet'
+    assert k_status['LastPVCRefAt']
+    assert k_status['LastPodRefAt']
+
+    # reuse that volume
+    p['pod_name'] = p['pod_name'].replace('kubernetes-status-test',
+                                          'kubernetes-status-test-reuse')
+    p['pvc_name'] = p['pvc_name'].replace('kubernetes-status-test',
+                                          'kubernetes-status-test-reuse')
+    p['pv_name'] = p['pvc_name']
+
+    csi_pv['metadata']['name'] = p['pv_name']
+    csi_pv['spec']['csi']['volumeHandle'] = volume_name
+    core_api.create_persistent_volume(csi_pv)
+
+    pvc['metadata']['name'] = p['pvc_name']
+    pvc['spec']['volumeName'] = p['pv_name']
+    core_api.create_namespaced_persistent_volume_claim(
+        body=pvc, namespace='default')
+
+    pod['metadata']['name'] = p['pod_name']
+    pod['spec']['volumes'] = [{
+        'name': pod['spec']['containers'][0]['volumeMounts'][0]['name'],
+        'persistentVolumeClaim': {
+            'claimName': p['pvc_name'],
+        },
+    }]
+    create_and_wait_pod(core_api, pod)
+
+    volume = client.by_id_volume(volume_name)
+    k_status = volume["kubernetesStatus"]
+    assert k_status['PVName'] == p['pv_name']
+    for i in range(RETRY_COMMAND_COUNT):
+        if k_status['PVStatus'] == 'Bound':
+            break
+        time.sleep(1)
+        volume = client.by_id_volume(volume_name)
+        k_status = volume["kubernetesStatus"]
+    assert k_status['PVStatus'] == 'Bound'
+    assert k_status['Namespace'] == 'default'
+    assert k_status['PVCName'] == p['pvc_name']
+    assert k_status['PodName'] == p['pod_name']
```

Comment:
This should be after the loop.

Reply:
Here I set `replicas=1` to speed up the test. Does 2 replicas for the statefulset matter?

Reply:
How much is the difference in speed? It's better to have more than one replicas since it's more likely the real use case.